src/core/gamemode_selected_event.h

```diff
@@ -0,0 +1,18 @@
+#ifndef INCLUDED_CORE_GAMEMODE_SELECTED_EVENT_H
+#define INCLUDED_CORE_GAMEMODE_SELECTED_EVENT_H
+
+#include "platform/event.h"
+
+namespace core {
+
+struct GamemodeSelectedEvent : public platform::Event
+{
+    std::string mGameMode;
+    GamemodeSelectedEvent(){}
+};
+
+} // namespace core
+
+#endif//INCLUDED_CORE_GAMEMODE_SELECTED_EVENT_H
+
+//command:  "./classgenerator" -g "event" -c "gamemode_selected_event"
```

Comment:
F: consider using -m "std::string-gameMode"
it will generate almost the same output like yours, but a more standardized event usage is encouraged. (will generate a string parametered ctor)